### PR TITLE
[cp-schema-regristry] add pdb + podAntiAffinity by default

### PR DESCRIPTION
**Pull request by aymeric-ledizes**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. helm upgrade on minikube, helm install on gke)

### CODE REVIEW

File: charts/cp-schema-registry/values.yaml [hunk 1]
```diff
@@ -93,9 +93,8 @@ nodeSelector: {}
 ## Ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
 tolerations: []
 
-## Pod scheduling constraints
-## Ref: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity
-affinity: {}
+podDisruptionBudget:
+  minAvailable: 1
```
Comment (frivoire) [inline charts/cp-schema-registry/values.yaml:97]: Max unavailable: 1   would be safer here

File: charts/cp-schema-registry/templates/deployment.yaml [hunk 2]
```diff
@@ -151,7 +151,17 @@ spec:
...
+                matchLabels:
+                  app: {{ .Release.Name }}
+              topologyKey: kubernetes.io/hostname
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - weight: 100
+              podAffinityTerm:
+                labelSelector:
+                  matchLabels:
+                    app: {{ .Release.Name }}
+                topologyKey: failure-domain.beta.kubernetes.io/zone
```
Comment (frivoire) [inline charts/cp-schema-registry/templates/deployment.yaml:167]: I think that this key is now deprecated. So still working, but we'll need to change all of those soon. You should try the new one (I don't remember it, sorry)

Comment (aymeric-ledizes) [reply charts/cp-schema-registry/templates/deployment.yaml]: The label still exist on nodes but `topology.kubernetes.io/zone` is the recommended now thanks :+1:

Comment (frivoire) [reply charts/cp-schema-registry/templates/deployment.yaml]: Yeah, it's the one I had in mind.
BTW: soon, we'll need to change on all our charts

**Pull request merged**